xnat_tools/bids_utils.py

```diff
@@ -65,38 +67,50 @@ def prepare_bids_prefixes(project, subject, session):
 
     return pi_prefix, study_prefix, subject_prefix, session_prefix
 
-def prepare_bids_output_path(bids_root_dir, pi_prefix, study_prefix, subject_prefix, session_prefix):
+def prepare_export_output_path(bids_root_dir, pi_prefix, study_prefix, subject_prefix, session_prefix, overwrite=False):
 
-    bids_study_dir = os.path.join(bids_root_dir, pi_prefix, study_prefix)
-    bids_subject_dir = os.path.join(bids_study_dir, "xnat-export", subject_prefix)
-    bids_session_dir = os.path.join(bids_subject_dir, session_prefix)
+    study_dir = os.path.join(bids_root_dir, pi_prefix, study_prefix)
+    subject_dir = os.path.join(study_dir, "xnat-export", subject_prefix)
+    session_dir = os.path.join(subject_dir, session_prefix)
 
     # Set up working directory
-    if not os.access(bids_session_dir, os.R_OK):
-        _logger.info('Making output BIDS Session directory %s' % bids_study_dir)
-        os.makedirs(bids_session_dir)
+    if overwrite and os.path.exists(session_dir):
+        _logger.info('Removing existing xnat-export session directory %s' % session_dir)
+        shutil.rmtree(session_dir, ignore_errors=True)
+
+    if not os.access(session_dir, os.R_OK):
+        _logger.info('Making output xnat-export session directory %s' % session_dir)
+        os.makedirs(session_dir)
 
-    return bids_session_dir
+    return session_dir
 
 def prepare_heudi_prefixes(project, subject, session):
     #get PI from project name
     pi_prefix = project.lower().split('_')[0] 
 
      # Paths to export source data in a BIDS friendly way
     study_prefix = "study-" + project.lower().split('_')[1]
-    subject_prefix = subject.lower().replace("_","")
-    session_prefix = session.lower().replace("_","")
+    subject_prefix = "sub-" + subject.lower().replace("_","")
+    session_prefix = "ses-"+ session.lower().replace("_","")
 
     return pi_prefix, study_prefix, subject_prefix, session_prefix
 
-def prepare_heudiconv_output_path(bids_root_dir, pi_prefix, study_prefix, subject_prefix, session_prefix):
+def prepare_heudiconv_output_path(bids_root_dir, pi_prefix, study_prefix, subject_prefix, session_prefix, overwrite=False):
 
     heudi_study_dir = os.path.join(bids_root_dir, pi_prefix, study_prefix)
     heudi_output_dir = os.path.join(heudi_study_dir, "bids")
+    subject_dir = os.path.join(heudi_output_dir, subject_prefix)
+    session_dir = os.path.join(subject_dir, session_prefix)
 
     # Set up working directory
+    if overwrite:
+        print("Overwrite set to True")
+        if os.path.exists(session_dir):
+            print('Removing existing heudi session directory %s' % session_dir)
+            shutil.rmtree(session_dir)
+
     if not os.access(heudi_output_dir, os.R_OK):
-        _logger.info('Making output BIDS Session directory %s' % heudi_output_dir)
+        print('Making output BIDS Session directory %s' % heudi_output_dir)
```

Comment:
That's a good question. If my memory is correct we don't have access to the logger in the heudiconv functions because the logger is taken by the hudiconv process. So I think that like want' being sent to STDOUT. I would need to explore how to make both loggers stream correctly. I'll make an issue for this one an tackle it later